### ISSUE TITLE
Import declaration conflict not detected with `import { Foo } from` statement

### ISSUE THREAD

**Issue by SamVerschueren** [Microsoft/TypeScript#16545]
**TypeScript Version:** 2.3.4

**Code**

```ts
export interface Foo {
  [key: string]: any;
}
```

```ts
import { Foo } from './foo';

const Foo = 'foo';
```

**Expected behavior:**

The compiler should fail with the following error

> [ts] Import declaration conflicts with local declaration of 'Foo'.

**Actual behavior:**

The compiler does not fail and allows this which leads to unexpected behaviour when running the code. However, it does throw when using the following import statement

```ts
import * as Foo from './foo';

const Foo = 'foo';
```

I created a short gif to illustrate the problem.

![tslint](https://user-images.githubusercontent.com/1913805/27166726-0cda463c-519d-11e7-972f-730a745bcbbc.gif)

**Comment by SamVerschueren** [comment]
I also noted that this only happens when you import an interface like that. If you import a class, for instance

```ts
export class Foo {
    bar = 'bar';
}
```

The compiler error is thrown. But it should throw with an interface as well, because when running the code, I receive the following runtime exception.

```
const Foo = 'foo';
            ^^^^^
SyntaxError: Identifier 'Foo' has already been declared
    at Object.exports.runInThisContext (vm.js:76:16)
    at Module._compile (module.js:542:28)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
``

**Comment by DanielRosenwasser** [comment]
@rbuckton gives a convincing argument that this is working appropriately because the imported `Foo` is only a type, so there's no conflict in the value space.

He also makes the argument that the `import * as Foo from "bar"` should actually be allowed because its usage might be elided.

**Comment by DanielRosenwasser** [comment]
To elaborate on the `class` example, classes always introduce a value at runtime.

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
